### PR TITLE
feat(tracer): Lightstep exposes new env var for clearing span buffer

### PR DESCRIPTION
**Pull request by dm03514**
refs #SREPR-580

## Summary

Updates lightstep tracer to set `clear_span_buffer_consecutive_errors` and expose it as an env var.  This should allow services to auto recover when flushing spans error because of payload size.

## Description
- https://github.com/lightstep/lightstep-tracer-javascript/commit/0626c1d9f1af5ad531dbf98cc9b97c87519009d9
- https://auth0team.atlassian.net/browse/SREPR-580
- https://auth0team.atlassian.net/browse/SREPR-579

## Checklist

- [ ] Has automated tests
- [ ] Compatible with Appliance
- [ ] Has no dependencies to other PRs
- [ ] Readme file updated to document changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "datadog-metrics": "~0.8.1",
     "gc-stats": "^1.4.0",
     "jaeger-client": "3.12.0",
-    "lightstep-tracer": "~0.24.2",
+    "lightstep-tracer": "lightstep/lightstep-tracer-javascript#0626c1d9f1af5ad531dbf98cc9b97c87519009d9",
```
Comment (dafortune) [inline package.json]: Isn't there are release that we point to a commit?

File: lib/tracer_factory.js [hunk 2]
```diff
@@ -35,7 +35,9 @@ exports.create = (agent, pkg, env, params) => {
       collector_port: env.TRACE_AGENT_PORT,
       collector_encryption: env.TRACE_AGENT_USE_TLS ? 'tls' : 'none',
       component_name: serviceName,
-      max_reporting_interval_millis: env.TRACE_REPORTING_INTERVAL_MILLIS || 500
+      max_reporting_interval_millis: env.TRACE_REPORTING_INTERVAL_MILLIS || 500,
+      // 5 results in 5 * max_reporting_interval_millis + jitter of full buffer = max(~5) seconds before flush
+      clear_span_buffer_consecutive_errors: env.LS_CLEAR_SPAN_BUFFER_AFTER_NUM_ERRORS || 5
```
Comment (dafortune) [inline lib/tracer_factory.js]: Do you think `max_span_records` is ok or should we also reduce it?

File: package.json (hunk 1 above)
Comment (dm03514) [reply package.json]: Asking ishmeet

File: lib/tracer_factory.js (hunk 2 above)
Comment (dm03514) [reply lib/tracer_factory.js]: updated to expose it, left it at 4096, but not opposed to changing default at all

File: package.json (hunk 1 above)
Comment (nw) [reply package.json]: I would rather see this published. They are generally pretty good at cutting releases quickly.

**Pull request merged**